File: 1224-minimum-falling-path-sum-ii/minimum-falling-path-sum-ii.py
class Solution:
    def minFallingPathSum(self, grid: List[List[int]]) -> int:
        n = len(grid)

        for r in range(1, n):
            for c in range(n):
                grid[r][c] += min(grid[r - 1][0:c] + grid[r - 1][c + 1:])

        return min(grid[-1])

File: 1224-minimum-falling-path-sum-ii/minimum-falling-path-sum-ii.java
class Solution {
    public int minFallingPathSum(int[][] grid) {
        int n = grid.length;

        for (int r = 1; r < n; r++) {
            int min1 = Integer.MAX_VALUE;
            int min2 = Integer.MAX_VALUE;
            for (int c = 0; c < n; c++) {
                if (min1 > grid[r - 1][c]) {
                    min2 = min1;
                    min1 = grid[r - 1][c];
                } else if (min2 > grid[r - 1][c]) {
                    min2 = grid[r - 1][c];
                }
            }
            
            for (int c = 0; c < n; c++) {
                if (grid[r - 1][c] == min1) {
                    grid[r][c] += min2;
                } else {
                    grid[r][c] += min1;
                }
            }
        }
        return Arrays.stream(grid[n - 1]).min().getAsInt();
    }
}